tests/threshold_tools/test_threshold_tools.py

```diff
@@ -0,0 +1,193 @@
+        test_data = test_data = np.random.rand(365 * 3) * 100
+        test = xr.DataArray(
+            data=test_data,
+            coords={
+                "time": pd.date_range(start="2001-01-01", end="2003-12-31"),
+            },
+        )
+        result = _calc_average_ess_timeseries_data(test, 1)
+        assert not np.isnan(result)
+        assert isinstance(result, float)
+
+    def test__get_fitted_distr(self):
+        """Check that fitted distr returns correct parameter set for gev."""
+        test_data = test_data = np.random.rand(10, 10, 365 * 3) * 100
+        test = xr.DataArray(
+            data=test_data,
+            coords={
+                "x": np.arange(10, 20, 1),
+                "y": np.arange(10, 20, 1),
+                "time": pd.date_range(start="2001-01-01", end="2003-12-31"),
+            },
+        )
+        result = _get_fitted_distr(test, "gev", scipy.stats.genextreme)
+        assert isinstance(result, tuple)
+        assert isinstance(result[0], dict)
+        assert isinstance(
+            result[1], scipy.stats._distn_infrastructure.rv_continuous_frozen
+        )
+        for param in ["c", "loc", "scale"]:
+            assert param in result[0]
+            assert isinstance(result[0][param], float)
+
+    def test__get_distr_func(self):
+        """Check that get_distr_func returns correct distribution for input value."""
+        diststr = ["gev", "gumbel", "weibull", "pearson3", "genpareto", "gamma"]
+        diststat = [
+            scipy.stats.genextreme,
+            scipy.stats.gumbel_r,
+            scipy.stats.weibull_min,
+            scipy.stats.pearson3,
+            scipy.stats.genpareto,
+            scipy.stats.gamma,
+        ]
+
+        for dstr, dstat in zip(diststr, diststat):
+            assert _get_distr_func(dstr) == dstat
+
+        with pytest.raises(ValueError):
+            _get_distr_func("")
+
+    def test__exceedance_count_name(self):
+        """Test that correct string is returned for various parameter values."""
+        item1 = 4
+        item2 = "day"
+        test = xr.DataArray()
+        test.attrs["duration2"] = (item1, item2)
+        result = _exceedance_count_name(test)
+        assert result == (f"Number of {item1}-{item2} events")
+
+        test = xr.DataArray()
+        test.attrs["duration2"] = None
+        test.attrs["group"] = (item1, item2)
+        result = _exceedance_count_name(test)
+        assert result == (f"Number of {item1}-{item2} events")
+
+        test = xr.DataArray()
+        test.attrs["duration2"] = None
+        test.attrs["group"] = None
+        tscales = {"hourly": "hours", "monthly": "months", "daily": "days"}
+        for item in tscales:
+            test.attrs["frequency"] = item
+            result = _exceedance_count_name(test)
+            assert result == (f"Number of {tscales[item]}")
+
+    def test__get_exceedance_events(self):
```

Comment:
Is there a reason all the tests in this class have "test__" with the double underscore in the name?

Reply:
ooo-- that makes sense! thanks!